### ISSUE TITLE
Plugin dependency locks aren't generated for plugin block

### ISSUE THREAD

**Issue by rgoldberg** [gradle/gradle#10437]
Plugin dependency locks (`buildscript-*.lockfile`) aren't generated, e.g., by the following Gradle Wrapper 5.6.1 command.

```
./gradlew dependencies --write-locks
```

for a `build.gradle.kts` with the following:

```
plugins {
    groovy
    idea
    `java-gradle-plugin`
    `java-library`
    id("com.gradle.plugin-publish") version "0.10.1"
}

buildscript {
    dependencyLocking.lockAllConfigurations()
}
```

Maybe it would work if I explicitly specified `buildscript {dependencies {…}}`, but I shouldn't have to do that.

I would imagine that only plugins that aren't part of the gradle distribution should have their dependencies locked.  I also imagine that neither Gradle itself, nor plugins that are distributed with Gradle, have transitive dependencies that must be supplied separately.  If that's wrong, then these transitive dependencies should be able to be locked.

**Comment by ljacomet** [comment]
This is caused by a surprising behavior from the Kotlin DSL:
`dependencyLocking.lockAllConfigurations()` is actually invoked on the `project`. There is no such method for the `buildscript` receiver. 

As [documented](https://docs.gradle.org/5.6.1/userguide/dependency_locking.html#locking_buildscript_classpath_configuration), you need the following to lock the buildscript classpath:
```
buildscript {
    configurations.classpath {
        resolutionStrategy.activateDependencyLocking()
    }
}
```

**Comment by rgoldberg** [comment]
@ljacomet: Sorry, I had read about locking plugin dependencies on my phone.  Once I got to my computer a while later & tried it, I just assumed that the syntax I tried was correct since it didn't produce any errors.

A few questions:

1. Is there any way to prevent either certain `project` members (like `dependencyLocking`), or all of them, from being invokable outside the root level?

2. If not, would it be useful to request that IntelliJ to visually indicate when members of a script root object are invoked outside of the root level?

3. It looks like dependency locking is only enforced for a configuration if it is activated for that configuration, and if a `.lockfile` exists for that configuration.  If that's correct, would it be useful to add a switch to generate an error if the `.lockfile` is missing for a configuration with activated locking?
